proj/funcDictionary.py

```diff
@@ -0,0 +1,335 @@
+            for more detail see documentation in plotData.py
+        i: int number of letters to be extracted from each strain, e.g. if i = 18,
+            then only the first 18 letters from each strain are extracted, the rest
+            is ignored
+        isExtractedNum: Bool, that says, if only a limited number of 
+            letters should be extracted; Thus, if isExtractNum == False, it makes no
+            sense, to set i > -1, because everything is extracted anyway.
+            If isExtractNum == True, i must be set to a certain value > -1.
+    output: a: list, containing all the loaded data
+        a4: list, containing the nucleic acid sequences
+        seq: list, containing the nucleic acid sequences first acids until 
+            the upper threshold "extractNum"
+        filename: string (either 'pre.txt' or 'post.txt')
+        ls: list of length values for each sequence in seq
+    """
+    # info: define filename
+    if step == 0:
+        filename = 'pre.txt'
+    if step == 1:
+        filename = 'post.txt'
+
+    # info: check, if plotData has an Ns-array, then choose N = plotData.Ns[i]; alternatively choose N = plotData.N[i]
+    if i > -1:
+        try: 
+            N = plotData.Ns[i]
+        except: 
+            print("\n Could not load element plotData.Ns[i]. Perhaps it does not exist?")
+            N = -1
+    else:
+        N = plotData.N
+
+    # info: make variables and lists as empty or zero objects
+    index = 0
+    a = list()
+    a4 = list()
+    seq = list()
+    ls = list() # right place to define ls?
+
+    # info: opening the list of generated sequences 
+    # info: (sequences generated done with SONIA, 
+    # info: which simulates VDJ recombination)
+    
+    with open(filename) as f:
+        for line in f.readlines():
+            if index < N:
+                a.append(line.split('\t'))
+                # info: attach the fourth element  to a4
+                a4.append(a[index][3].replace('\n', ''))
+                if isExtractNum == True:
+                    seq.append(a4[index][:plotData.extractNum])
+                else:
+                    seq.append(a4[index][:])
+                ls.append(len(seq[index]))
+                
+                index += 1
+    return a, a4, seq, filename, ls
+
+def make_sub_graphs(G):
+    """
+    info: make list of connected subgraphs of G (subgraph with the highest number of nodes) 
+    input: G: graph
+    output: sub_graphs: list of subgraphs, sorted according to size
+    """
+    # info: make sorted list of subgraphs
+    sub_graphs = sorted(list(G.subgraph(c) for c in nx.connected_components(G)), key = len, reverse = True)
+
+
+    # info: if sub_graphs is empty , make a base graph
+    if len(sub_graphs) == 0:
+        sub_graphs = list()
+        sub_graphs.append(nx.Graph())
+        sub_graphs[0].add_node('FillNode')
+
+    return sub_graphs
+
+def biggest_sub_graph(sub_graphs):
```

Comment:
`largest_sub_graph`

This whole function should be replaced, because it's not very efficient for determining the largest connected component. Please also don't use the term "subgraph" if you mean connected component. A subgraph can be connected, but it could also be a set of isolated nodes.

Simply use this function `Gc = max(nx.connected_component_subgraphs(G), key=len)` to determine the largest connected component.

See https://networkx.github.io/documentation/networkx-1.10/reference/generated/networkx.algorithms.components.connected.connected_component_subgraphs.html.